src/main/java/htsjdk/variant/vcf/VCFHeader.java

```diff
@@ -68,7 +68,7 @@
     private final Map<String, VCFFormatHeaderLine> mFormatMetaData = new LinkedHashMap<String, VCFFormatHeaderLine>();
     private final Map<String, VCFFilterHeaderLine> mFilterMetaData = new LinkedHashMap<String, VCFFilterHeaderLine>();
     private final Map<String, VCFHeaderLine> mOtherMetaData = new LinkedHashMap<String, VCFHeaderLine>();
-    private final List<VCFContigHeaderLine> contigMetaData = new ArrayList<VCFContigHeaderLine>();
+    private final Map<String, VCFContigHeaderLine> contigMetaData = new LinkedHashMap<>();
```

Comment:
In general I agree with you about this, but using a LinkedHashMap here is actually an API restriction.  If we change to using a different Map that doesn't provide the ordering properties of LinkedHashMap then this class will no longer work.  It would be ideal if java provided an InsertionOrderedMap interface, but since it doesn't it's probably best to specify the LinkedHashMap here even though it's bad practice in general to declare a specific type.